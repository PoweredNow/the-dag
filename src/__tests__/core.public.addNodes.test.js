const addNodes = require('../core/public/addNodes');
const { getState } = require('../core/private/_state');
describe('core.public.addNodes', () => {
  test('exports', () => {
    expect(addNodes).toMatchSnapshot();
  });
  test('Adds Single Node', () => {
    const node = { nodeID: 1, data: { some: 'data' } };
    addNodes([node]);
    expect(getState()).toMatchSnapshot();
  });

  test('Adds Two Nodes', () => {
    const sourceNode = { nodeID: 1, data: { some: 'data' } };
    const targetNode = { nodeID: 2, data: { some: 'otherData' } };
    addNodes([sourceNode, targetNode]);
    expect(getState()).toMatchSnapshot();
  });
  test('Fails silently with no arguments', () => {
    addNodes();
    const state = getState();
    expect(state).toMatchSnapshot();
  });
});
