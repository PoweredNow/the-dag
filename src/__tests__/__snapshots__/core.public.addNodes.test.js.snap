// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core.public.addNodes Adds Single Node 1`] = `
Object {
  "edges": Object {},
  "nodes": Object {
    "1": Object {
      "nodeData": Object {
        "some": "data",
      },
      "nodeID": 1,
      "possibleTargets": Array [],
    },
  },
}
`;

exports[`core.public.addNodes Adds Two Nodes 1`] = `
Object {
  "edges": Object {},
  "nodes": Object {
    "1": Object {
      "nodeData": Object {
        "some": "data",
      },
      "nodeID": 1,
      "possibleTargets": Array [],
    },
    "2": Object {
      "nodeData": Object {
        "some": "otherData",
      },
      "nodeID": 2,
      "possibleTargets": Array [],
    },
  },
}
`;

exports[`core.public.addNodes Fails silently with no arguments 1`] = `
Object {
  "edges": Object {},
  "nodes": Object {
    "1": Object {
      "nodeData": Object {
        "some": "data",
      },
      "nodeID": 1,
      "possibleTargets": Array [],
    },
    "2": Object {
      "nodeData": Object {
        "some": "otherData",
      },
      "nodeID": 2,
      "possibleTargets": Array [],
    },
  },
}
`;

exports[`core.public.addNodes exports 1`] = `[Function]`;
